com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -527,7 +527,12 @@ private void NetworkVariableUpdate(ulong clientId, int behaviourIndex)
             {
                 for (int k = 0; k < NetworkVariableFields.Count; k++)
                 {
-                    NetworkManager.SnapshotSystem.Store(NetworkObjectId, behaviourIndex, k, NetworkVariableFields[k]);
+                    var update = new UpdateCommand();
```

Comment:
This path misses/requires a `ShouldWrite` test, otherwise update will be spammed for variables that didn't change.